src/core/client.ts

```diff
@@ -101,76 +94,52 @@ export class BTCPayClient {
 
   private async signed_get_request(
     path: string,
-    params: any,
-    token?: any,
+    params: any = {},
+    token: any = _.values(this.tokens)[0],
   ): Promise<any> {
-    const _token = token ? token : _.values(this.tokens)[0];
-    const _params = params ? params : {};
-    _params['token'] = _token;
+    params.token = token;
 
-    const _uri = this.host + path;
-    const _payload = '?' + qs.stringify(_params);
     const _options = JSON.parse(JSON.stringify(this.options));
 
+    const _uri = this.host + path;
+    const _payload = '?' + qs.stringify(params);
+
     _.extend(_options.headers, this.create_signed_headers(_uri, _payload));
-    _options['uri'] = _uri;
-    _options['qs'] = _params;
-
-    try {
-      const resp = await rp.get(_options);
-      return resp['data'];
-    } catch (err) {
-      throw err;
-    }
+    _options.uri = _uri;
+    _options.qs = params;
+
+    return rp.get(_options).then((resp: any) => resp.data);
   }
 
   private async signed_post_request(
     path: string,
-    payload: any,
-    token: any,
+    payload: any = {},
+    token: any = _.values(this.tokens)[0],
```

Comment:
Can you really assign a function execution as a default parameter value?

Reply:
yes. see the compiled JavaScript :-D